presto-clp/src/main/java/com/facebook/presto/plugin/clp/ClpMetadataFilterProvider.java

```diff
@@ -0,0 +1,221 @@
+ * filters based on a JSON configuration file.
+ * <p>
+ * The configuration file locates at {@code clp.metadata-filter-config} and defines metadata filters
+ * for different scopes:
+ * <ul>
+ *   <li><b>Catalog-level</b>: applies to all schemas and tables under the catalog.</li>
+ *   <li><b>Schema-level</b>: applies to all tables under the specified catalog and schema.</li>
+ *   <li><b>Table-level</b>: applies to the fully qualified catalog.schema.table.</li>
+ * </ul>
+ * <p>
+ * Each scope maps to a list of filter definitions. Each filter includes:
+ * <ul>
+ *   <li>{@code filterName}: must match a column name in the table's schema. Note that only numeric
+ *   type column can be used as metadata filter now.</li>
+ *   <li>{@code rangeMapping} (optional): specifies how the filter should be remapped when it
+ *   targets metadata-only columns. Note that this option is valid only if the column is numeric
+ *   type.
+ *       For example, a condition like {@code "msg.timestamp" > 1234 AND "msg.timestamp" < 5678}
+ *       will be rewritten as {@code end_timestamp > 1234 AND begin_timestamp < 5678} to ensure
+ *       metadata-based filtering produces a superset of the actual result.</li>
+ * </ul>
+ * <p>
+ * This provider is used by {@code ClpFilterToKqlConverter} to determine which columns are eligible
+ * for metadata filter push down, and by {@code ClpMySqlSplitProvider} to construct metadata filter
+ * queries that determine which splits to read.
+ */
+public class ClpMetadataFilterProvider
+{
+    private final Map<String, TableConfig> filterMap;
+
+    @Inject
+    public ClpMetadataFilterProvider(ClpConfig config)
+    {
+        requireNonNull(config, "config is null");
+
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            filterMap = mapper.readValue(
+                    new File(config.getMetadataFilterConfig()),
+                    new TypeReference<Map<String, TableConfig>>() {});
+        }
+        catch (IOException e) {
+            throw new PrestoException(CLP_METADATA_FILTER_CONFIG_NOT_FOUND, "Failed to metadata filter config file open.");
+        }
+    }
+
+    public Map<String, TableConfig> getFilterMap()
+    {
+        return filterMap;
+    }
+
+    public void checkContainsAllFilters(SchemaTableName schemaTableName, String metadataFilterKqlQuery)
+    {
+        boolean hasAllMetadataFilterColumns = true;
+        String notFoundFilterColumnName = "";
+        for (String columnName : getFilterNames(format("%s.%s", CONNECTOR_NAME, schemaTableName))) {
+            if (!metadataFilterKqlQuery.contains(columnName)) {
+                hasAllMetadataFilterColumns = false;
+                notFoundFilterColumnName = columnName;
+                break;
+            }
+        }
+        if (!hasAllMetadataFilterColumns) {
+            throw new PrestoException(CLP_MANDATORY_METADATA_FILTER_NOT_VALID, notFoundFilterColumnName + " is a mandatory metadata filter column but not valid");
```

Comment:
If it's necessary to contain all filters, should we have a list of `notFoundFilterColumnName`